### PR TITLE
Sanitize param usage in some plumbing task script blocks

### PR DESCRIPTION
**Pull request by sbwsg**
# Changes

Prior to this commit we had several script blocks using the syntax
`$(params.foo)` to pull in param values. As per https://github.com/tektoncd/plumbing/issues/971
it is advised to pull these usages into environment variables to
improve robustness (unexpected characters in param values like quotes and ampersands
can cause confusing issues when a script runs) and to remove
one avenue of script injection in scenarios when a param includes
user-submitted content.

This PR patches resources in the cd, images, nightly-release,
nightly-tests and release/base directories to move param referencing
into environment variables.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Review by jerop — APPROVED** [review]
thanks @sbwsg!

File: tekton/resources/nightly-tests/base/deploy_tekton_component.yaml [hunk 1]
```diff
@@ -12,6 +12,13 @@ spec:
   - name: extra-file
     description: extra file to install (for instance, interceptors.yaml for triggers)
     default: ""
+  env:
+  - name: PACKAGE
+    value: $(params.package)
+  - name: VERSION
+    value: $(params.version)
+  - name: EXTRA_FILE
+    value: $(params.extra-file)
```
Comment (afrittoli) [inline tekton/resources/nightly-tests/base/deploy_tekton_component.yaml:21]: You cannot have an `env` directly under `spec`, it must either go into `steps` or into a `stepTemplate`.

File: tekton/resources/nightly-tests/bastion-p/k8s_cluster_setup.yaml [hunk 2]
```diff
@@ -33,13 +33,22 @@ spec:
...
       default: create
+  env:
+    - name: REMOTE_HOST
+      value: $(params.remote-host)
+    - name: REMOTE_USER
+      value: $(params.remote-user)
+    - name: REMOTE_PORT
+      value: $(params.remote-port)
+    - name: ACTION
+      value: $(params.action)
```
Comment (afrittoli) [inline tekton/resources/nightly-tests/bastion-p/k8s_cluster_setup.yaml:44]: Ditto

File: tekton/resources/release/base/prerelease_checks.yaml [hunk 3]
```diff
@@ -25,6 +25,13 @@ spec:
   - name: releaseBucket
     description: >-
       The bucket where to look for the release, in the format gs://<bucket-name>/<project-name>
+  env:
+  - name: PACKAGE
+    value: $(params.package)
+  - name: VERSION_TAG
+    value: $(params.versionTag)
+  - name: RELEASE_BUCKET
+    value: $(params.releaseBucket)
```
Comment (afrittoli) [inline tekton/resources/release/base/prerelease_checks.yaml:34]: ditto

**Pull request merged**